Adapted from the following patch.

From 5c2b84b0963aa1af0c55c38afbc4080fd9f6a46f Mon Sep 17 00:00:00 2001
From: Jon Roelofs <jonathan_roelofs@apple.com>
Date: Fri, 17 Nov 2023 14:21:57 -0800
Subject: [PATCH] [builtins] Move cfi start's after the symbol name [NFC]

... in preparation for diagnosing improperly nested .cfi regions.

See https://reviews.llvm.org/D155245
---
 compiler-rt/lib/builtins/assembly.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/compiler-rt/lib/builtins/assembly.h b/compiler-rt/lib/builtins/assembly.h
index f6ce6a9fccff..f859ad8f6706 100644
--- a/lib/builtins/assembly.h
+++ b/lib/builtins/assembly.h
@@ -249,9 +249,10 @@
   .globl name SEPARATOR                                                        \
   SYMBOL_IS_FUNC(name) SEPARATOR                                               \
   DECLARE_SYMBOL_VISIBILITY(name) SEPARATOR                                    \
-  CFI_START SEPARATOR                                                          \
   DECLARE_FUNC_ENCODING                                                        \
-  name: SEPARATOR BTI_C
+  name:                                                                        \
+  SEPARATOR CFI_START                                                          \
+  SEPARATOR BTI_C
 
 #define DEFINE_COMPILERRT_FUNCTION_ALIAS(name, target)                         \
   .globl SYMBOL_NAME(name) SEPARATOR                                           \
-- 
2.44.0

